### ISSUE TITLE
SetAutoDelete Rebus in unrecoverable state after queue deleted

### ISSUE THREAD

**Issue by pjh1974** [rebus-org/Rebus.RabbitMq#26]
I am working on an application using Rebus and RabbitMQ in which client applications subscribe to messages on the service bus.  

To ensure that client application's queues are cleaned-up when the client closes, I configure the RabbitMQ transport with SetAutoDelete(true).  This works perfectly for cleaning up the queue but has unwanted side-effects if the client is paused in the debugger for > 1 minute or the client becomes disconnected from the network for >1 minute.  I can understand why I am getting this behaviour - it seems that RabbitMQ is destroying the queue when it can no longer see the client.  This is OK but Rebus doesn't seem to be able to recover from this situation when client activity is restored.  Furthermore, the client goes into a seemingly infinite state of the logging the following:

```
Exception thrown: 'System.IO.FileLoadException' in System.Messaging.dll
Rebus.RabbitMq.ConnectionManager: Information: Existing connection found to be CLOSED
The thread 0x5fdc has exited with code 0 (0x0).
Exception thrown: 'System.NullReferenceException' in Evolve.Logging.dll
Exception thrown: 'System.IO.FileLoadException' in System.Messaging.dll
Rebus.RabbitMq.RabbitMqTransport: Warning: Could not initialize consumer: RabbitMQ.Client.Exceptions.OperationInterruptedException: The AMQP operation was interrupted: AMQP close-reason, initiated by Peer, code=404, text="NOT_FOUND - no queue 'RebusTestClient_FRF-L-11642[18748]' in vhost '/'", classId=60, methodId=20, cause=
   at RabbitMQ.Client.Impl.SimpleBlockingRpcContinuation.GetReply(TimeSpan timeout)
   at RabbitMQ.Client.Impl.ModelBase.BasicConsume(String queue, Boolean autoAck, String consumerTag, Boolean noLocal, Boolean exclusive, IDictionary`2 arguments, IBasicConsumer consumer)
   at RabbitMQ.Client.Impl.AutorecoveringModel.BasicConsume(String queue, Boolean autoAck, String consumerTag, Boolean noLocal, Boolean exclusive, IDictionary`2 arguments, IBasicConsumer consumer)
   at RabbitMQ.Client.IModelExensions.BasicConsume(IModel model, String queue, Boolean autoAck, IBasicConsumer consumer)
   at Rebus.RabbitMq.RabbitMqTransport.InitializeConsumer()
   at Rebus.RabbitMq.RabbitMqTransport.EnsureConsumerInitialized() - waiting 2 seconds
```

I'm not sure if there is a better way to achieve my desired behaviour but it doesn't seem quite right that the client ties-up a whole thread continually trying to perform an operation that will never succeed and then fill the log in the process.

**Comment by pjh1974** [comment]
@mookid8000 I'm just about to submit a PR that will add functionality to restore the queue and any subscriptions.  I'm not sure if this is the right thing to do or not - but it seems to have the desired effect for me.

**Comment by mookid8000** [comment]
Fixed by #27
